pkg/registry/server/server.go

```diff
@@ -328,8 +329,29 @@ func ResolveAndMergeConfigsAndInjectTest(configs Getter, resolver Resolver, reso
 				})
 			}
 			for step, resources := range config.Resources {
-				if step == "*" { // * is special, and we will only pull from one config to merge, this is "last in wins" we may need to do better in the future
-					mergedConfig.Resources[step] = resources
+				if step == "*" { // * is special, and the ref should not be appended, it will be merged to use the greatest value instead
+					if existing, ok := mergedConfig.Resources["*"]; ok {
+						replaceIfGreater := func(resourceType string) {
+							existingValue, err := resource.ParseQuantity(existing.Requests[resourceType])
+							if err != nil {
+								logger.WithError(err).Warnf("couldn't parse existing '%s' resource quantity", resourceType)
+							}
+							value, err := resource.ParseQuantity(resources.Requests[resourceType])
+							if err != nil {
+								logger.WithError(err).Warnf("couldn't parse '%s' resource quantity", resourceType)
+							}
+							if existingValue.Cmp(value) < 0 { // This value is higher than existing
```

Comment:
What happens here if one of the previous two `ParseQuantity` fails?

Reply:
Ah that would likely lead to a stack trace. It _shouldn't_ ever happen, but we know how that goes. I will add a check around that.